src/java.base/share/classes/java/io/Reader.java

```diff
@@ -140,6 +140,125 @@ public void close() {
         };
     }
 
+    /**
+     * Returns a {@code Reader} that reads characters from a
+     * {@code CharSequence}, starting at the first character in the sequence.
+     *
+     * <p> The resulting reader is not safe for use by multiple
+     * concurrent threads. If the reader is to be used by more than one
+     * thread it should be controlled by appropriate synchronization.
+     *
+     * <p> If the sequence changes while the reader is open, e.g. the length
+     * changes, the behavior is undefined.
+     *
+     * <p> The returned reader is initially open. The reader is closed by
+     * calling the {@code close()} method. Subsequent calls to {@code close()}
+     * have no effect. After the reader has been closed, the {@code read()},
+     * {@code read(char[])}, {@code read(char[], int, int)},
+     * {@code read(CharBuffer)}, {@code ready()}, {@code skip(long)}, and
+     * {@code transferTo()} methods all throw {@code IOException}.
+     *
+     * <p> The returned reader supports the {@link #mark mark()} operation.
+     *
+     * @param cs {@code CharSequence} providing the character stream.
+     *
+     * @return a {@code Reader} which reads characters from {@code cs}
+     *
+     * @throws NullPointerException if {@code cs} is {@code null}
+     *
+     * @since 24
+     */
+    public static Reader of(final CharSequence cs) {
+        Objects.requireNonNull(cs);
+
+        return new Reader() {
+            private boolean isClosed;
+            private int next = 0;
+            private int mark = 0;
+
+            /** Check to make sure that the stream has not been closed */
+            private void ensureOpen() throws IOException {
+                if (isClosed)
+                    throw new IOException("Stream closed");
```

Comment:
Should the exception text instead be "Reader closed" and also the code comment a couple of lines above, replace "stream" to "reader"?

Reply:
The complete implementation was copied from "StringReader", so it was "Stream closed" there already, and "StringReader" now refers to this new code location. I deliberately did not modify anything else but the essentially needed parts. If we change this text here, this might be unexpected by existing applications.

Reply:
Okay, it's reasonable to keep it in this form then.